### PR TITLE
Support for avatar decoration

### PR DESCRIPTION
**Pull request by Puncher1**
## Summary

This PR adds support for avatar decoration.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/user.py [hunk 1]
```diff
@@ -175,6 +179,18 @@ def display_avatar(self) -> Asset:
         """
         return self.avatar or self.default_avatar
 
+    @property
```
Comment (bijij) [inline discord/user.py:182]: this should be cached.

Comment (Puncher1) [reply discord/user.py:182]: Why?

Comment (bijij) [reply discord/user.py:182]: On thinking again you're right, my mistake.
Brain fart.

Comment (Puncher1) [reply discord/user.py:182]: 👍

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/asset.py [hunk 2]
```diff
@@ -246,6 +246,15 @@ def _from_guild_avatar(cls, state: _State, guild_id: int, member_id: int, avatar
             animated=animated,
         )
 
+    @classmethod
+    def _from_avatar_decoration(cls, state: _State, avatar_decoration: str) -> Self:
+        return cls(
+            state,
+            url=f'{cls.BASE}/avatar-decoration-presets/{avatar_decoration}.png?size=96',
```
[the remark this answers is not in the document]
Comment (Puncher1) [reply discord/asset.py]: I just went by the other assets who have a hardcoded size, or am I misunderstanding something?

[the remark this answers is not in the document]
Comment (Rapptz) [reply discord/asset.py]: Ah. My mistake, the URL is parsed so it's actually fine.

**Comment by Rapptz** [comment]
> I have observed on few users profiles, the avatar decoration asset URL is in the format of `https://cdn.discordapp.com/avatar-decorations/{user_id}/{avatar_decoration_hash}.png` probably for legacy or very early released avatar decos, havent dig too deep into it. Is it in the scope of this PR to address those cases?

This is the actual documented URL in the in-progress PR so it should be the one used.

**Comment by Rapptz** [comment]
If you have time please rebase this PR to the 2.4 release now that the docs PR is merged.

**Comment by Puncher1** [comment]
Done

**Comment by Soheab** [comment]
Looks like the field is going to be changed or has already and returns an object now with the `sku_id` and `asset`
https://github.com/discord/discord-api-docs/pull/6464

File: discord/member.py [hunk 3]
```diff
@@ -464,18 +470,19 @@ def _presence_update(self, data: PartialPresenceUpdate, user: UserPayload) -> Op
 
     def _update_inner_user(self, user: UserPayload) -> Optional[Tuple[User, User]]:
         u = self._user
-        original = (u.name, u.discriminator, u._avatar, u.global_name, u._public_flags)
+        original = (u.name, u.discriminator, u._avatar, u.global_name, u._public_flags, u._avatar_decoration_data)
```
Comment (Rapptz) [inline discord/member.py:473]: dict comparisons like this are a bit iffy, we should probably compare a key in the dict instead

File: discord/user.py [hunk 4]
```diff
@@ -187,6 +188,30 @@ def display_avatar(self) -> Asset:
...
+    def avatar_decoration_sku_id(self) -> Optional[int]:
+        """Optional[:class:`int`]: Returns the SKU ID of the avatar decoration the user has.
+
+        If the user has not set an avatar decoration, ``None`` is returned.
+
+        .. versionadded:: 2.4
+        """
+        if self._avatar_decoration_data is not None:
+            return int(self._avatar_decoration_data['sku_id'])
+        return None
```
Comment (Rapptz) [inline discord/user.py:213]:
```suggestion
        return _get_as_snowflake(self._avatar_decoration_data, 'sku_id')
```

Requires importing it above from `utils`.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/member.py (hunk 3 above)
Comment (Puncher1) [reply discord/member.py]: It's checked with the `sku_id` now and if it differs from the original, the new dict is assigned.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**